druid/src/widget/textbox.rs

```diff
@@ -265,20 +255,21 @@ impl Widget<String> for TextBox {
             .with_save(|rc| {
                 rc.clip(clip_rect);
 
+                // Calculate layout
+                let text_layout = self.get_layout(rc.text(), data, env);
+
                 // Shift everything inside the clip by the hscroll_offset
                 rc.transform(Affine::translate((-self.hscroll_offset, 0.)));
 
                 // Draw selection rect
                 if !self.selection.is_caret() {
                     let (left, right) = (self.selection.min(), self.selection.max());
 
-                    let selection_width =
-                        self.substring_measurement_hack(rc.text(), data, left, right, env);
+                    let selection_width = self.x_for_offset(&text_layout, right)
+                        - self.x_for_offset(&text_layout, left);
 
                     let selection_pos = Point::new(
-                        self.substring_measurement_hack(rc.text(), data, 0, left, env)
-                            + PADDING_LEFT
-                            - 1.,
+                        self.x_for_offset(&text_layout, left) + PADDING_LEFT - 1.,
```

Comment:
total nit but since we use this twice I would probably just do a `let left_offset = self.x_for_offset(&text_layout, left)` above, and use both places, which probably let this fit on one line. 🤷‍♂